### DISCUSSION TITLE
Abilitity to set remoteEnv with initializeCommand

### DISCUSSION THREAD

**Discussion by davidbarratt** [devcontainers/community#54] [Ideas]
We currently have a `compose.yml` that is like this:
```yml
services:
  dev:
    environment:
      NPM_TOKEN: ${NPM_TOKEN}
```
where the value of the `NPM_TOKEN` enivornment variable is taken from the developers current terminal.

The way this is populated, is with a script that looks something like this:
```sh
aws sso login;
export NPM_TOKEN=$(aws secretsmanager get-secret-value --secret-id npm-token);
```

When putting this in an `initializeCommand` it seems the value is not exported into the same shell that is used by devcontainers, meaning docker compose sets it to an empty string:
```
WARN[0000] The "NPM_TOKEN" variable is not set. Defaulting to a blank string. 
```

It seems like it is impossible for us to safely use devcontainers until this is fixed. :/ 

One way to fix this would be to be able to use the same shell for the script as where the `docker compose up` command is run.

There could be other solutions too, like running an `initContainer` to startup a container that can set environment variables or something. I'm open to other solutions!

**Comment by bamurtaugh** [comment]
Thank you for opening this @davidbarratt!

@chrmarti @alexander-smolyakov I'd be interested in getting your thoughts on this one.
